src/app/core/store/documents/documents.filters.ts

```diff
@@ -18,50 +18,89 @@
 import {DocumentModel} from './document.model';
+import {QueryConverter} from "../navigation/query.converter";
 
-export class DocumentsFilters {
+export function filterDocumentsByQuery(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
+  documents = documents.filter(document => typeof(document) === 'object');
+  if (!query) {
+    return documents;
+  }
+
+  let filteredDocuments = filterDocumentsByCollections(documents, query);
+  filteredDocuments = filterDocumentsByFulltext(filteredDocuments, query);
+  filteredDocuments = filterDocumentsByFilters(filteredDocuments, query);
 
-  public static filterByQuery(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
-    documents = documents.filter(document => typeof(document) === 'object');
-    if (!query) {
-      return documents;
-    }
+  return paginate(filteredDocuments, query);
+}
 
-    let filteredDocuments = this.filterByCollections(documents, query);
-    filteredDocuments = this.filterByFulltext(filteredDocuments, query);
-    return this.paginate(filteredDocuments, query);
-  };
+function filterDocumentsByCollections(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
+  if (!hasCollectionsFilter(query)) {
+    return documents;
+  }
 
-  private static filterByCollections(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
-    if (!this.hasCollectionsFilter(query)) {
-      return documents;
-    }
+  return documents.filter(document => {
+    return query.collectionIds.includes(document.collectionId);
+  });
+}
 
-    return documents.filter(document => {
-      return query.collectionIds.includes(document.collectionId);
-    });
+function filterDocumentsByFulltext(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
+  if (!query.fulltext) {
+    return documents;
   }
 
-  private static hasCollectionsFilter(query: QueryModel): boolean {
-    return query.collectionIds && query.collectionIds.length > 0;
+  return documents.filter(document => Object.values(document.data).some(value => value.includes(query.fulltext)));
+}
+
+function filterDocumentsByFilters(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
+  if (!query.filters || query.filters.length == 0) {
+    return documents;
   }
 
-  private static filterByFulltext(documents: DocumentModel[], query: QueryModel): DocumentModel[] {
-    if (!query.fulltext) {
-      return documents;
-    }
+  const filters = query.filters.map(filter => QueryConverter.parseFilter(filter))
+    .filter(filter => !isNullOrUndefined(filter));
 
-    return documents.filter(document => Object.values(document.data).some(value => value.includes(query.fulltext)));
+  return documents.filter(document => documentMeetsFilters(document, filters));
+}
+
+function documentMeetsFilters(document: DocumentModel, filters: AttributeFilter[]): boolean {
+  let meets = true;
+  filters.forEach(filter => meets = meets && documentMeetFilter(document, filter));
+  return meets;
```

Comment:
```typescript
return filters.every(filter => documentMeetFilter(document, filter));
```

Reply:
I was looking for something like that, thanks!